[2025-10-18T16:30:47.339+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: test_hdfs_spark.read_hdfs_with_spark manual__2025-10-18T16:30:36.093608+00:00 [queued]>
[2025-10-18T16:30:48.315+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: test_hdfs_spark.read_hdfs_with_spark manual__2025-10-18T16:30:36.093608+00:00 [queued]>
[2025-10-18T16:30:48.316+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-10-18T16:30:48.538+0000] {taskinstance.py:2191} INFO - Executing <Task(SparkSubmitOperator): read_hdfs_with_spark> on 2025-10-18 16:30:36.093608+00:00
[2025-10-18T16:30:48.579+0000] {standard_task_runner.py:60} INFO - Started process 4090 to run task
[2025-10-18T16:30:48.601+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'test_hdfs_spark', 'read_hdfs_with_spark', 'manual__2025-10-18T16:30:36.093608+00:00', '--job-id', '100', '--raw', '--subdir', 'DAGS_FOLDER/test_hdfs_spark.py', '--cfg-path', '/tmp/tmpldk1e71r']
[2025-10-18T16:30:48.611+0000] {standard_task_runner.py:88} INFO - Job 100: Subtask read_hdfs_with_spark
[2025-10-18T16:30:48.817+0000] {task_command.py:423} INFO - Running <TaskInstance: test_hdfs_spark.read_hdfs_with_spark manual__2025-10-18T16:30:36.093608+00:00 [running]> on host 2ef039423968
[2025-10-18T16:30:49.248+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='test_hdfs_spark' AIRFLOW_CTX_TASK_ID='read_hdfs_with_spark' AIRFLOW_CTX_EXECUTION_DATE='2025-10-18T16:30:36.093608+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-18T16:30:36.093608+00:00'
[2025-10-18T16:30:49.352+0000] {base.py:83} INFO - Using connection ID 'spark_default' for task execution.
[2025-10-18T16:30:49.355+0000] {spark_submit.py:341} INFO - Spark-Submit cmd: spark-submit --master local --name arrow-spark --verbose --queue default /opt/***/dags/hdfs_spark_test.py
[2025-10-18T16:31:15.134+0000] {spark_submit.py:492} INFO - Using properties file: null
[2025-10-18T16:31:18.005+0000] {spark_submit.py:492} INFO - Parsed arguments:
[2025-10-18T16:31:18.036+0000] {spark_submit.py:492} INFO - master                  local
[2025-10-18T16:31:19.071+0000] {spark_submit.py:492} INFO - remote                  null
[2025-10-18T16:31:19.086+0000] {spark_submit.py:492} INFO - deployMode              null
[2025-10-18T16:31:19.094+0000] {spark_submit.py:492} INFO - executorMemory          null
[2025-10-18T16:31:19.170+0000] {spark_submit.py:492} INFO - executorCores           null
[2025-10-18T16:31:19.190+0000] {spark_submit.py:492} INFO - totalExecutorCores      null
[2025-10-18T16:31:19.205+0000] {spark_submit.py:492} INFO - propertiesFile          null
[2025-10-18T16:31:19.256+0000] {spark_submit.py:492} INFO - driverMemory            null
[2025-10-18T16:31:19.274+0000] {spark_submit.py:492} INFO - driverCores             null
[2025-10-18T16:31:19.356+0000] {spark_submit.py:492} INFO - driverExtraClassPath    null
[2025-10-18T16:31:19.360+0000] {spark_submit.py:492} INFO - driverExtraLibraryPath  null
[2025-10-18T16:31:19.371+0000] {spark_submit.py:492} INFO - driverExtraJavaOptions  null
[2025-10-18T16:31:19.372+0000] {spark_submit.py:492} INFO - supervise               false
[2025-10-18T16:31:19.373+0000] {spark_submit.py:492} INFO - queue                   default
[2025-10-18T16:31:19.374+0000] {spark_submit.py:492} INFO - numExecutors            null
[2025-10-18T16:31:19.455+0000] {spark_submit.py:492} INFO - files                   null
[2025-10-18T16:31:19.475+0000] {spark_submit.py:492} INFO - pyFiles                 null
[2025-10-18T16:31:19.477+0000] {spark_submit.py:492} INFO - archives                null
[2025-10-18T16:31:19.479+0000] {spark_submit.py:492} INFO - mainClass               null
[2025-10-18T16:31:19.490+0000] {spark_submit.py:492} INFO - primaryResource         file:/opt/***/dags/hdfs_spark_test.py
[2025-10-18T16:31:19.495+0000] {spark_submit.py:492} INFO - name                    arrow-spark
[2025-10-18T16:31:19.505+0000] {spark_submit.py:492} INFO - childArgs               []
[2025-10-18T16:31:19.555+0000] {spark_submit.py:492} INFO - jars                    null
[2025-10-18T16:31:19.572+0000] {spark_submit.py:492} INFO - packages                null
[2025-10-18T16:31:19.587+0000] {spark_submit.py:492} INFO - packagesExclusions      null
[2025-10-18T16:31:19.602+0000] {spark_submit.py:492} INFO - repositories            null
[2025-10-18T16:31:19.657+0000] {spark_submit.py:492} INFO - verbose                 true
[2025-10-18T16:31:19.663+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:19.664+0000] {spark_submit.py:492} INFO - Spark properties used, including those specified through
[2025-10-18T16:31:19.665+0000] {spark_submit.py:492} INFO - --conf and those from the properties file null:
[2025-10-18T16:31:19.666+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:19.667+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:19.686+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:26.155+0000] {spark_submit.py:492} INFO - Main class:
[2025-10-18T16:31:26.158+0000] {spark_submit.py:492} INFO - org.apache.spark.deploy.PythonRunner
[2025-10-18T16:31:26.171+0000] {spark_submit.py:492} INFO - Arguments:
[2025-10-18T16:31:26.172+0000] {spark_submit.py:492} INFO - file:/opt/***/dags/hdfs_spark_test.py
[2025-10-18T16:31:26.179+0000] {spark_submit.py:492} INFO - null
[2025-10-18T16:31:26.184+0000] {spark_submit.py:492} INFO - Spark config:
[2025-10-18T16:31:26.185+0000] {spark_submit.py:492} INFO - (spark.app.name,arrow-spark)
[2025-10-18T16:31:26.190+0000] {spark_submit.py:492} INFO - (spark.app.submitTime,1760805085889)
[2025-10-18T16:31:26.192+0000] {spark_submit.py:492} INFO - (spark.master,local)
[2025-10-18T16:31:26.268+0000] {spark_submit.py:492} INFO - (spark.submit.deployMode,client)
[2025-10-18T16:31:26.269+0000] {spark_submit.py:492} INFO - (spark.submit.pyFiles,)
[2025-10-18T16:31:26.270+0000] {spark_submit.py:492} INFO - Classpath elements:
[2025-10-18T16:31:26.271+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:26.272+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:26.273+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:31:37.519+0000] {spark_submit.py:492} INFO - Starting Spark HDFS Test...
[2025-10-18T16:31:39.564+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:39 INFO SparkContext: Running Spark version 3.5.1
[2025-10-18T16:31:39.760+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:39 INFO SparkContext: OS info Linux, 6.6.87.2-microsoft-standard-WSL2, amd64
[2025-10-18T16:31:39.882+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:39 INFO SparkContext: Java version 17.0.16
[2025-10-18T16:31:43.295+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:43 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-10-18T16:31:48.160+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:48 INFO ResourceUtils: ==============================================================
[2025-10-18T16:31:48.181+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:48 INFO ResourceUtils: No custom resources configured for spark.driver.
[2025-10-18T16:31:48.195+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:48 INFO ResourceUtils: ==============================================================
[2025-10-18T16:31:48.196+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:48 INFO SparkContext: Submitted application: HDFS-Test
[2025-10-18T16:31:51.663+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:51 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2025-10-18T16:31:51.789+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:51 INFO ResourceProfile: Limiting resource is cpu
[2025-10-18T16:31:51.856+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:51 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2025-10-18T16:31:54.567+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:54 INFO SecurityManager: Changing view acls to: ***
[2025-10-18T16:31:54.690+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:54 INFO SecurityManager: Changing modify acls to: ***
[2025-10-18T16:31:54.768+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:54 INFO SecurityManager: Changing view acls groups to:
[2025-10-18T16:31:54.786+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:54 INFO SecurityManager: Changing modify acls groups to:
[2025-10-18T16:31:54.814+0000] {spark_submit.py:492} INFO - 25/10/18 16:31:54 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2025-10-18T16:32:04.176+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:04 INFO Utils: Successfully started service 'sparkDriver' on port 43981.
[2025-10-18T16:32:06.062+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:06 INFO SparkEnv: Registering MapOutputTracker
[2025-10-18T16:32:09.844+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:09 INFO SparkEnv: Registering BlockManagerMaster
[2025-10-18T16:32:10.593+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:10 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2025-10-18T16:32:10.757+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:10 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2025-10-18T16:32:10.786+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:10 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2025-10-18T16:32:12.765+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:12 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-da64aca3-fa0e-4051-8ecb-1dbb6d1b9dbf
[2025-10-18T16:32:13.334+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:13 INFO MemoryStore: MemoryStore started with capacity 413.9 MiB
[2025-10-18T16:32:14.284+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:14 INFO SparkEnv: Registering OutputCommitCoordinator
[2025-10-18T16:32:23.360+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:23 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2025-10-18T16:32:30.512+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:30 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2025-10-18T16:32:38.451+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:38 INFO Executor: Starting executor ID driver on host 2ef039423968
[2025-10-18T16:32:38.888+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:38 INFO Executor: OS info Linux, 6.6.87.2-microsoft-standard-WSL2, amd64
[2025-10-18T16:32:39.137+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:38 INFO Executor: Java version 17.0.16
[2025-10-18T16:32:39.391+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:39 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2025-10-18T16:32:39.438+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:39 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@40cb28e4 for default.
[2025-10-18T16:32:40.470+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:40 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 44333.
[2025-10-18T16:32:40.542+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:40 INFO NettyBlockTransferService: Server created on 2ef039423968:44333
[2025-10-18T16:32:40.572+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:40 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2025-10-18T16:32:41.242+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:41 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2ef039423968, 44333, None)
[2025-10-18T16:32:41.462+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:41 INFO BlockManagerMasterEndpoint: Registering block manager 2ef039423968:44333 with 413.9 MiB RAM, BlockManagerId(driver, 2ef039423968, 44333, None)
[2025-10-18T16:32:41.534+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:41 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2ef039423968, 44333, None)
[2025-10-18T16:32:41.652+0000] {spark_submit.py:492} INFO - 25/10/18 16:32:41 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2ef039423968, 44333, None)
[2025-10-18T16:33:02.175+0000] {spark_submit.py:492} INFO - Trying to read from: hdfs://192.168.1.10:9000/user/winardi/data/olist/olist_customers_dataset.csv
[2025-10-18T16:33:06.627+0000] {spark_submit.py:492} INFO - 25/10/18 16:33:06 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2025-10-18T16:33:07.114+0000] {spark_submit.py:492} INFO - 25/10/18 16:33:07 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2025-10-18T16:34:21.371+0000] {spark_submit.py:492} INFO - 25/10/18 16:34:21 INFO AsyncEventQueue: Process of event SparkListenerExecutorMetricsUpdate(driver,WrappedArray(),Map()) by listener AppStatusListener took 10.182269241s.
[2025-10-18T16:34:55.200+0000] {spark_submit.py:492} INFO - 25/10/18 16:34:55 WARN FileStreamSink: Assume no metadata directory. Error while looking for metadata directory in the path: hdfs://192.168.1.10:9000/user/winardi/data/olist/olist_customers_dataset.csv.
[2025-10-18T16:34:55.258+0000] {spark_submit.py:492} INFO - java.net.ConnectException: Call From 2ef039423968/172.18.0.6 to hadoop:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[2025-10-18T16:34:55.259+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-10-18T16:34:55.260+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-10-18T16:34:55.261+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-10-18T16:34:55.423+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-10-18T16:34:55.458+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-10-18T16:34:55.518+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:913)
[2025-10-18T16:34:55.537+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:828)
[2025-10-18T16:34:55.538+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1616)
[2025-10-18T16:34:55.539+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1558)
[2025-10-18T16:34:55.623+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1455)
[2025-10-18T16:34:55.717+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:242)
[2025-10-18T16:34:55.725+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:129)
[2025-10-18T16:34:55.729+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy35.getFileInfo(Unknown Source)
[2025-10-18T16:34:55.734+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:965)
[2025-10-18T16:34:55.735+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:34:55.741+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:34:55.822+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:34:55.918+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:34:55.933+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
[2025-10-18T16:34:56.019+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
[2025-10-18T16:34:56.031+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
[2025-10-18T16:34:56.032+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
[2025-10-18T16:34:56.034+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
[2025-10-18T16:34:56.047+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy36.getFileInfo(Unknown Source)
[2025-10-18T16:34:56.050+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1739)
[2025-10-18T16:34:56.119+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1753)
[2025-10-18T16:34:56.133+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1750)
[2025-10-18T16:34:56.139+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
[2025-10-18T16:34:56.141+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1765)
[2025-10-18T16:34:56.224+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystem.isDirectory(FileSystem.java:1777)
[2025-10-18T16:34:56.231+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.streaming.FileStreamSink$.hasMetadata(FileStreamSink.scala:54)
[2025-10-18T16:34:56.233+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:366)
[2025-10-18T16:34:56.234+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2025-10-18T16:34:56.246+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2025-10-18T16:34:56.248+0000] {spark_submit.py:492} INFO - at scala.Option.getOrElse(Option.scala:189)
[2025-10-18T16:34:56.249+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2025-10-18T16:34:56.250+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.DataFrameReader.csv(DataFrameReader.scala:538)
[2025-10-18T16:34:56.252+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:34:56.253+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:34:56.255+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:34:56.318+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:34:56.327+0000] {spark_submit.py:492} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-10-18T16:34:56.329+0000] {spark_submit.py:492} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-10-18T16:34:56.333+0000] {spark_submit.py:492} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2025-10-18T16:34:56.335+0000] {spark_submit.py:492} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-10-18T16:34:56.346+0000] {spark_submit.py:492} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-10-18T16:34:56.418+0000] {spark_submit.py:492} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-10-18T16:34:56.437+0000] {spark_submit.py:492} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-10-18T16:34:56.443+0000] {spark_submit.py:492} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2025-10-18T16:34:56.445+0000] {spark_submit.py:492} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-10-18T16:34:56.522+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-10-18T16:34:56.527+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-10-18T16:34:56.530+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-10-18T16:34:56.532+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:205)
[2025-10-18T16:34:56.538+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:586)
[2025-10-18T16:34:56.617+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:711)
[2025-10-18T16:34:56.622+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:833)
[2025-10-18T16:34:56.626+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:414)
[2025-10-18T16:34:56.627+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getConnection(Client.java:1677)
[2025-10-18T16:34:56.628+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1502)
[2025-10-18T16:34:56.629+0000] {spark_submit.py:492} INFO - ... 40 more
[2025-10-18T16:35:03.781+0000] {spark_submit.py:492} INFO - ERROR: An error occurred while calling o29.csv.
[2025-10-18T16:35:03.799+0000] {spark_submit.py:492} INFO - : java.net.ConnectException: Call From 2ef039423968/172.18.0.6 to hadoop:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[2025-10-18T16:35:03.803+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-10-18T16:35:03.805+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-10-18T16:35:03.881+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-10-18T16:35:03.882+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-10-18T16:35:03.899+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-10-18T16:35:03.900+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:913)
[2025-10-18T16:35:03.969+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:828)
[2025-10-18T16:35:03.976+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1616)
[2025-10-18T16:35:03.985+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1558)
[2025-10-18T16:35:04.163+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1455)
[2025-10-18T16:35:04.165+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:242)
[2025-10-18T16:35:04.166+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:129)
[2025-10-18T16:35:04.167+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy35.getFileInfo(Unknown Source)
[2025-10-18T16:35:04.283+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:965)
[2025-10-18T16:35:04.301+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:35:04.369+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:35:04.473+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:35:04.622+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:35:04.907+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
[2025-10-18T16:35:04.910+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
[2025-10-18T16:35:04.969+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
[2025-10-18T16:35:04.971+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
[2025-10-18T16:35:04.986+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
[2025-10-18T16:35:05.078+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy36.getFileInfo(Unknown Source)
[2025-10-18T16:35:05.082+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1739)
[2025-10-18T16:35:05.101+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1753)
[2025-10-18T16:35:05.111+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1750)
[2025-10-18T16:35:05.176+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
[2025-10-18T16:35:05.197+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1765)
[2025-10-18T16:35:05.269+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystem.exists(FileSystem.java:1760)
[2025-10-18T16:35:05.283+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4(DataSource.scala:756)
[2025-10-18T16:35:05.284+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4$adapted(DataSource.scala:754)
[2025-10-18T16:35:05.285+0000] {spark_submit.py:492} INFO - at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:384)
[2025-10-18T16:35:05.288+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-10-18T16:35:05.289+0000] {spark_submit.py:492} INFO - at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-10-18T16:35:05.289+0000] {spark_submit.py:492} INFO - at scala.util.Success.map(Try.scala:213)
[2025-10-18T16:35:05.296+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-10-18T16:35:05.376+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-10-18T16:35:05.378+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-10-18T16:35:05.379+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-10-18T16:35:05.380+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
[2025-10-18T16:35:05.381+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[2025-10-18T16:35:05.382+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[2025-10-18T16:35:05.383+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[2025-10-18T16:35:05.384+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[2025-10-18T16:35:05.389+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[2025-10-18T16:35:05.674+0000] {spark_submit.py:492} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-10-18T16:35:05.691+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-10-18T16:35:05.882+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-10-18T16:35:05.974+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-10-18T16:35:06.170+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:205)
[2025-10-18T16:35:06.280+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:586)
[2025-10-18T16:35:06.281+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:711)
[2025-10-18T16:35:06.283+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:833)
[2025-10-18T16:35:06.289+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:414)
[2025-10-18T16:35:06.291+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getConnection(Client.java:1677)
[2025-10-18T16:35:06.369+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1502)
[2025-10-18T16:35:06.386+0000] {spark_submit.py:492} INFO - ... 37 more
[2025-10-18T16:35:06.389+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:35:06.400+0000] {spark_submit.py:492} INFO - Traceback (most recent call last):
[2025-10-18T16:35:06.401+0000] {spark_submit.py:492} INFO - File "/opt/***/dags/hdfs_spark_test.py", line 20, in main
[2025-10-18T16:35:06.403+0000] {spark_submit.py:492} INFO - df = spark.read.csv(hdfs_path, header=True, inferSchema=True)
[2025-10-18T16:35:06.405+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 740, in csv
[2025-10-18T16:35:06.415+0000] {spark_submit.py:492} INFO - return self._df(self._jreader.csv(self._spark._sc._jvm.PythonUtils.toSeq(path)))
[2025-10-18T16:35:06.416+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-10-18T16:35:06.477+0000] {spark_submit.py:492} INFO - return_value = get_return_value(
[2025-10-18T16:35:06.486+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2025-10-18T16:35:06.494+0000] {spark_submit.py:492} INFO - return f(*a, **kw)
[2025-10-18T16:35:06.501+0000] {spark_submit.py:492} INFO - File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2025-10-18T16:35:06.569+0000] {spark_submit.py:492} INFO - raise Py4JJavaError(
[2025-10-18T16:35:06.579+0000] {spark_submit.py:492} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o29.csv.
[2025-10-18T16:35:06.584+0000] {spark_submit.py:492} INFO - : java.net.ConnectException: Call From 2ef039423968/172.18.0.6 to hadoop:9000 failed on connection exception: java.net.ConnectException: Connection refused; For more details see:  http://wiki.apache.org/hadoop/ConnectionRefused
[2025-10-18T16:35:06.598+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2025-10-18T16:35:07.105+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2025-10-18T16:35:07.487+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2025-10-18T16:35:07.498+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
[2025-10-18T16:35:07.574+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
[2025-10-18T16:35:07.595+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapWithMessage(NetUtils.java:913)
[2025-10-18T16:35:07.671+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.wrapException(NetUtils.java:828)
[2025-10-18T16:35:07.702+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1616)
[2025-10-18T16:35:07.781+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1558)
[2025-10-18T16:35:07.890+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1455)
[2025-10-18T16:35:07.903+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:242)
[2025-10-18T16:35:07.969+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.ProtobufRpcEngine2$Invoker.invoke(ProtobufRpcEngine2.java:129)
[2025-10-18T16:35:07.995+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy35.getFileInfo(Unknown Source)
[2025-10-18T16:35:08.009+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.getFileInfo(ClientNamenodeProtocolTranslatorPB.java:965)
[2025-10-18T16:35:08.087+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-10-18T16:35:08.170+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-10-18T16:35:08.286+0000] {spark_submit.py:492} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-10-18T16:35:08.301+0000] {spark_submit.py:492} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-10-18T16:35:08.374+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:422)
[2025-10-18T16:35:08.470+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:165)
[2025-10-18T16:35:08.571+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:157)
[2025-10-18T16:35:08.687+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:95)
[2025-10-18T16:35:08.692+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:359)
[2025-10-18T16:35:08.718+0000] {spark_submit.py:492} INFO - at jdk.proxy2/jdk.proxy2.$Proxy36.getFileInfo(Unknown Source)
[2025-10-18T16:35:08.774+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DFSClient.getFileInfo(DFSClient.java:1739)
[2025-10-18T16:35:08.973+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1753)
[2025-10-18T16:35:09.187+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem$29.doCall(DistributedFileSystem.java:1750)
[2025-10-18T16:35:09.218+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystemLinkResolver.resolve(FileSystemLinkResolver.java:81)
[2025-10-18T16:35:09.270+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.hdfs.DistributedFileSystem.getFileStatus(DistributedFileSystem.java:1765)
[2025-10-18T16:35:09.372+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.fs.FileSystem.exists(FileSystem.java:1760)
[2025-10-18T16:35:09.419+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4(DataSource.scala:756)
[2025-10-18T16:35:09.422+0000] {spark_submit.py:492} INFO - at org.apache.spark.sql.execution.datasources.DataSource$.$anonfun$checkAndGlobPathIfNecessary$4$adapted(DataSource.scala:754)
[2025-10-18T16:35:09.472+0000] {spark_submit.py:492} INFO - at org.apache.spark.util.ThreadUtils$.$anonfun$parmap$2(ThreadUtils.scala:384)
[2025-10-18T16:35:09.571+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
[2025-10-18T16:35:09.586+0000] {spark_submit.py:492} INFO - at scala.util.Success.$anonfun$map$1(Try.scala:255)
[2025-10-18T16:35:09.601+0000] {spark_submit.py:492} INFO - at scala.util.Success.map(Try.scala:213)
[2025-10-18T16:35:09.870+0000] {spark_submit.py:492} INFO - at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[2025-10-18T16:35:09.885+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[2025-10-18T16:35:09.900+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-10-18T16:35:09.905+0000] {spark_submit.py:492} INFO - at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-10-18T16:35:09.913+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
[2025-10-18T16:35:09.981+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
[2025-10-18T16:35:09.999+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
[2025-10-18T16:35:10.079+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
[2025-10-18T16:35:10.172+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
[2025-10-18T16:35:10.186+0000] {spark_submit.py:492} INFO - at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[2025-10-18T16:35:10.202+0000] {spark_submit.py:492} INFO - Caused by: java.net.ConnectException: Connection refused
[2025-10-18T16:35:10.470+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnect(Native Method)
[2025-10-18T16:35:10.491+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
[2025-10-18T16:35:10.503+0000] {spark_submit.py:492} INFO - at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
[2025-10-18T16:35:10.577+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.SocketIOWithTimeout.connect(SocketIOWithTimeout.java:205)
[2025-10-18T16:35:10.596+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.net.NetUtils.connect(NetUtils.java:586)
[2025-10-18T16:35:10.598+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupConnection(Client.java:711)
[2025-10-18T16:35:10.599+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.setupIOstreams(Client.java:833)
[2025-10-18T16:35:10.601+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client$Connection.access$3800(Client.java:414)
[2025-10-18T16:35:10.691+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.getConnection(Client.java:1677)
[2025-10-18T16:35:10.698+0000] {spark_submit.py:492} INFO - at org.apache.hadoop.ipc.Client.call(Client.java:1502)
[2025-10-18T16:35:10.790+0000] {spark_submit.py:492} INFO - ... 37 more
[2025-10-18T16:35:10.883+0000] {spark_submit.py:492} INFO - 
[2025-10-18T16:35:11.178+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:03 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2025-10-18T16:35:11.211+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:04 INFO SparkUI: Stopped Spark web UI at http://2ef039423968:4040
[2025-10-18T16:35:11.277+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:06 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2025-10-18T16:35:11.287+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:07 INFO MemoryStore: MemoryStore cleared
[2025-10-18T16:35:11.289+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:07 INFO BlockManager: BlockManager stopped
[2025-10-18T16:35:11.290+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:08 INFO BlockManagerMaster: BlockManagerMaster stopped
[2025-10-18T16:35:11.290+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:08 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2025-10-18T16:35:11.291+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:09 INFO SparkContext: Successfully stopped SparkContext
[2025-10-18T16:35:12.787+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:12 INFO ShutdownHookManager: Shutdown hook called
[2025-10-18T16:35:12.819+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:12 INFO ShutdownHookManager: Deleting directory /tmp/spark-9b863973-542d-4c39-87e8-1385b0de392a
[2025-10-18T16:35:13.112+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-d0e1104b-aa1c-4944-ba3f-7bf4f905374e
[2025-10-18T16:35:13.398+0000] {spark_submit.py:492} INFO - 25/10/18 16:35:13 INFO ShutdownHookManager: Deleting directory /tmp/spark-d0e1104b-aa1c-4944-ba3f-7bf4f905374e/pyspark-4c7ca220-c905-464e-bd63-0e48c5b22cd8
[2025-10-18T16:35:24.377+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 423, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master local --name arrow-spark --verbose --queue default /opt/***/dags/hdfs_spark_test.py. Error code is: 1.
[2025-10-18T16:35:25.369+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=test_hdfs_spark, task_id=read_hdfs_with_spark, execution_date=20251018T163036, start_date=20251018T163047, end_date=20251018T163524
[2025-10-18T16:35:34.988+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 100 for task read_hdfs_with_spark (Cannot execute: spark-submit --master local --name arrow-spark --verbose --queue default /opt/***/dags/hdfs_spark_test.py. Error code is: 1.; 4090)
[2025-10-18T16:35:36.297+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-10-18T16:35:37.283+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
